SuffolkLITLab/courtformsonline.org
# 18 new jurisdictions via config

https://github.com/SuffolkLITLab/courtformsonline.org/issues/18

Adds ability to add jurisdictions by modifying the formsources.config file.

Changes:
- refactored main page component
- broke topic cards into their own component
- set up topic section and data query to fetch data based on jurisdiction
- set build up for dynamic routes for paths like `/ma` and `/ma/housing`
- add path data to config file
- Moves fetch functions higher in hierarchy to be at page level
- adds pathToServer config which allows assignment of servers to a given path
- Adjusts link tag behavior that that serverUrl is assigned to a given interview
 
Testing
- Navigate to homepage and not that the default suffolk data is there
- Naviagate to localhost:3000/gb and observe that the greater boston data is rendered
- Click card backgrounds and observer that the correct topic placeholders render
- Click tags and verify that correct app url is linked to
- Verify that tests/build work
- Note that the build output reflects the new pages

Build output log:
<img width="690" alt="image" src="https://github.com/SuffolkLITLab/courtformsonline.org/assets/61624970/f7f429d1-e0b6-4d93-a13b-5231764386bd">

Greater boston cards at /gb:
<img width="1340" alt="image" src="https://github.com/SuffolkLITLab/courtformsonline.org/assets/61624970/79f144fc-84f5-4fc4-9125-e4726df0d255">

 Placeholder component for topics at new route:
<img width="562" alt="image" src="https://github.com/SuffolkLITLab/courtformsonline.org/assets/61624970/f444825e-75a4-41dd-bad2-28b6b38ec941">


Note:  Also some changes to images in an effort to get them working on Pages. I dont have a way to test pages deployment locally so there may be some trial and error figuring out what the issue is there
